addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -1141,6 +1141,102 @@ QUnit.test('warning on send with shortcut when attempting to post message with s
     );
 });
 
+QUnit.test('remove an uploading attachment', async function (assert) {
+    assert.expect(4);
+
+    await this.start({
+        async mockFetch(resource, init) {
+            const res = this._super(...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await new Promise(() => {});
+            }
+            return res;
+        }
+    });
+    const composer = this.env.models['mail.composer'].create();
+    await this.createComposerComponent(composer);
+    const file = await createFile({
+        content: 'hello, world',
+        contentType: 'text/plain',
+        name: 'text.txt',
+    });
+    await afterNextRender(() =>
+        inputFiles(
+            document.querySelector('.o_FileUploader_input'),
+            [file]
+        )
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Composer_attachmentList',
+        "should have an attachment list"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Composer .o_Attachment',
+        "should have only one attachment"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Composer .o_Attachment.o-temporary',
+        "should have a temporary attachment"
```

Comment:
prefer using "uploading" in UI than "temporary"

Reply:
Btw why not also in code? It's much more explicit.

Reply:
it should also be in code, but I don't want to increase scope of PR too much.

Reply:
FYI task-2337268
https://www.odoo.com/web#id=2337268&action=3206&active_id=1519&model=project.task&view_type=form&cids=1&menu_id=4720